### ISSUE TITLE
Inconsistent URLs in action links using initial install of ver 2.4.129

### ISSUE THREAD

**Issue by chrisinmtown** [MISP/MISP#6163]
I followed the installation instructions on Ubuntu server 20.04 and the web server came up the first time, kudos for a solid install script.  

To meet the prereqs, I created a new server VM with VirtualBox on a laptop. At start the MISP server listens by default on localhost:8443 within that VM.  But to view the web pages, I set the VM network interface to bridged, then accessed the MISP server endpoint from the host laptop, which meant using a URL like https://192.168.1.14

This report is for some inconsistencies in the URLs/links on the UI.  When viewing an event, the left menu bar includes these items:

- "Edit event" with the URL https://localhost:8443/events/edit/79.
- "Delete event" with the URL https://192.168.1.14/events/delete/79#

As a result I cannot use the Edit link but I *can* use the Delete link.

It looks like the method of specifying the action URL is different for each.  The Edit action is given as a hyperlink ("a" element) where the HREF attribute has an absolute URL complete with protocol etc., in other words "https://localhost.../blah/blah".  But the Delete action is a tiny form with the action attribute (relative URL) of just "/events/delete/79" which the browser helpfully extends and makes it work. 

Ideally all the links should be relative URLs, and none should specify or assume the hostname and port?  I need to dig in to see if this is a configuration option at install time.

I'll attach screenshots where I've hovered over the Edit and Delete actions in the hope that a picture is worth 1,000 words here :)

Thanks for listening

<img width="1008" alt="misp-delete-event-good-url" src="https://user-images.githubusercontent.com/10234212/89061797-58831580-d333-11ea-8b0e-1f5a208dd8cc.png">
<img width="1008" alt="misp-edit-event-bad-url" src="https://user-images.githubusercontent.com/10234212/89061799-59b44280-d333-11ea-9d83-352d60dd196d.png">

*If you would like to report a bug, please fill the template bellow*

### Work environment

| Questions                 | Answers
|---------------------------|--------------------
| Type of issue             | Bug
| OS version (server)       |  Ubuntu 20.04
| OS version (client)       | MacOS
| PHP version               | not sure -- whatever comes with 2.4.129
| MISP version / git hash   | 2.4.129
| Browser                   | Firefox 79.0

### Expected behavior

All the buttons/actions/links on the page should work.

### Actual behavior

Some buttons/actions/links have appropriate URLs and work, others have inappropriate URLs and do not work.

### Steps to reproduce the behavior

Access the server via a base URL other than localhost:8443

### Logs, screenshots, configuration dump, ...

Please see attached images.

Glad to provide any more info that I missed.

**Comment by github-germ** [comment]
May help... What do you see for `MISP.baseurl` in WebUI `Administration > Server Settings & Maintenance` in the `MISP settings` (i.e. `.../servers/serverSettings/MISP`)?

**Comment by chrisinmtown** [comment]
Thanks @github-germ for pointing me to the settings page, here's what I see:

<img width="886" alt="Screen Shot 2020-07-31 at 1 50 59 PM" src="https://user-images.githubusercontent.com/10234212/89062686-e8758f00-d334-11ea-979f-a15b80de86fc.png">

**Comment by github-germ** [comment]
Does changing `MISP.baseurl` to your `https://192.168.1.14` help? I'm not saying this is the ideal solution based on how you did the install, but it might fix this _after-the-fact_.

**Comment by chrisinmtown** [comment]
Yes thank you, changing MISP.baseurl made things work *in that scenario*.  I'm sure it broke other things that I have not yet found.  

Will you leave this open as a request to harmonize how URLs are constructed and shown in the UI?

**Comment by github-germ** [comment]
> Will you leave this open

I'm a user like you, not in the CIRCL teasm. So yes, I will leave it open as I cannot close it :-) Seems reasonable.

**Comment by gimmic** [comment]
Fresh install to Ubuntu 20.04 LTS shows this symptom. 
> This is an initial install Powered by MISP 2.4.131 Please configure and harden accordingly - 2020-09-11 13:45:07

**Comment by iglocska** [comment]
Generally the baseurl is used to prepend all links - if you find any that don't adhere to it please let us know and we'll fix them - will have a look at the edit vs delete buttons

**Comment by chrisinmtown** [comment]
@iankoetter this has gone stale, can you post a link that's bad in your install? Or should I just close this?
